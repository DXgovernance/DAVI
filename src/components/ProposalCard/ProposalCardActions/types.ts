import { UseProposalVotesOfVoterReturn } from 'Modules/Guilds/Hooks/useProposalVotesOfVoter';

export interface ProposalCardActionsProps {
  votesOfVoter: UseProposalVotesOfVoterReturn;
  proposalCreator: string;
  userAddress: string;
}
