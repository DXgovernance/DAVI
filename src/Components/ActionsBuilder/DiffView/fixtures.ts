export const mockCodeOriginal = `
  {
    "colors": {
      "text": "red",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "lightGrey": "#BDC0C7",
        "grey": "orange"
      },
    }
  }
`;

export const mockCodeUpdated = `
  {
    "colors": {
      "text": "#fff",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "grey": "#A1A6B0",
        "lightGrey": "#BDC0C7"
      },
    }
  }
`;
