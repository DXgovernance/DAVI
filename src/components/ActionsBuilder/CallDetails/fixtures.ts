import {
  ApproveSendTokens,
  DecodedCall,
  SupportedAction,
} from 'components/ActionsBuilder/types';
import ERC20 from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20.abi);

export const emptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const decodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0x95a223299319022a842d0dfe4851c145a2f615b9',
  value: BigNumber.from(0),
  args: {
    _to: '0x0000000000000000000000000000000000000000',
    _value: BigNumber.from(1000),
  },
};

export const approvetokenSpendingMock: ApproveSendTokens = {
  callType: SupportedAction.GENERIC_CALL,
  from: '0x0000000000000000000000000000000000000000',
  to: '0x0000000000000000000000000000000000000000',
  value: BigNumber.from(0),
  function: null,
  args: {},
  amount: BigNumber.from(1000),
  token: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
};
