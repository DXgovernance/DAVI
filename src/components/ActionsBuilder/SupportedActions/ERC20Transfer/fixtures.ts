import { DecodedCall, SupportedAction } from 'components/ActionsBuilder/types';
import ERC20ABI from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20ABI.abi);

export const erc20TransferEmptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const erc20TransferDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0xfFb1cd0F95368DDd06D556161c5D3d9f0f4Fe6d2',
  value: BigNumber.from(0),
  args: {
    _to: '0xaF8eB8C3A5d9d900AA0B98e3Df0bcC17d3C5F698',
    _value: BigNumber.from('5000000000000000000'),
  },
};
