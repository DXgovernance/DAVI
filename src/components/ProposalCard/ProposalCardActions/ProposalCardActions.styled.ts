import styled from 'styled-components';

export const ProposalCardActionsWrapper = styled.div`
  width: 5rem;
  display: flex;
  justify-content: space-between;
`;
