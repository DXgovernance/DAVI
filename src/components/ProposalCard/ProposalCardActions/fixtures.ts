import { BigNumber } from 'ethers';

export const mockProposalCardActionsProps = {
  votesOfVoters: {
    action: '1',
    votingPower: BigNumber.from(0),
  },
  proposalCreated: true,
  userAddress: '0x0000000000000000000000000000000000000000',
};
