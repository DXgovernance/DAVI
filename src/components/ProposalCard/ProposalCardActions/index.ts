export { default as ProposalCardActions } from './ProposalCardActions';
