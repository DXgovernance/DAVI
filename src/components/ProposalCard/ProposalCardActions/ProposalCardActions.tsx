import { IoMdCheckmarkCircleOutline } from 'react-icons/io';
import { HiOutlinePencil } from 'react-icons/hi';
import { ProposalCardActionsWrapper } from './ProposalCardActions.styled';

const ProposalCardActions = ({
  votesOfVoters,
  proposalCreated,
  userAddress,
}) => {
  const isCreator = proposalCreated === userAddress;
  const hasVoted =
    votesOfVoters?.action !== null && votesOfVoters?.votingPower !== null;
  return (
    <ProposalCardActionsWrapper>
      {hasVoted ? <IoMdCheckmarkCircleOutline size={30} /> : null}
      {isCreator ? <HiOutlinePencil size={30} /> : null}
    </ProposalCardActionsWrapper>
  );
};

export default ProposalCardActions;
