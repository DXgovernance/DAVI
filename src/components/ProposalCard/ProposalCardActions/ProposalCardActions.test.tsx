import ProposalCardActions from './ProposalCardActions';
import { render } from 'utils/tests';
import { mockProposalCardActionsProps } from './fixtures';

describe('ProposalCardActions', () => {
  it('renders properly and shows two icons if you have created the proposal and voted', () => {
    const { container } = render(
      <ProposalCardActions {...mockProposalCardActionsProps} />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders properly and shows no icons if you have neither created the proposal nor voted', () => {
    const mockProps = {
      ...mockProposalCardActionsProps,
      votesOfVoter: {
        action: null,
        votingPower: null,
        proposalCreated: false,
      },
    };
    const { container } = render(<ProposalCardActions {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
