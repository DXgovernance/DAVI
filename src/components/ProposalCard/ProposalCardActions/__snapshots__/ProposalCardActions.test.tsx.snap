// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProposalCardActions renders properly and shows no icons if you have neither created the proposal nor voted 1`] = `
.c0 {
  width: 5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div>
   
  <div
    class="c0"
  />
   
</div>
`;

exports[`ProposalCardActions renders properly and shows two icons if you have created the proposal and voted 1`] = `
.c0 {
  width: 5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div>
   
  <div
    class="c0"
  >
    <svg
      fill="currentColor"
      height="30"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M170.718 216.482L141.6 245.6l93.6 93.6 208-208-29.118-29.118L235.2 279.918l-64.482-63.436zM422.4 256c0 91.518-74.883 166.4-166.4 166.4S89.6 347.518 89.6 256 164.482 89.6 256 89.6c15.6 0 31.2 2.082 45.764 6.241L334 63.6C310.082 53.2 284.082 48 256 48 141.6 48 48 141.6 48 256s93.6 208 208 208 208-93.6 208-208h-41.6z"
      />
    </svg>
  </div>
   
</div>
`;
